pkg/apis/config/v1alpha1/resmgr/policy/balloons/config.go

```diff
@@ -149,6 +149,12 @@ type BalloonDef struct {
 	// this will be the number of CPUs reserved for it even if a container
 	// would request less.
 	MinCpus int `json:"minCPUs,omitempty"`
+	// MemoryTypes lists memory types allowed to containers in a
+	// balloon. Supported types are: DRAM, HBM, PMEM. By default
+	// all memory types in the system are allowed.
+	// +kubebuilder:validation:MaxItems=3
+	// +kubebuilder:validation:items:Pattern=`^(DRAM|HBM|PMEM)$`
+	MemoryTypes []string `json:"memoryTypes,omitempty"`
```

Comment:
@klihub have you verified this?

because I just deployed it a cluster and tried to add non DRAM|HBM|PMEM values and validation didn't complain
```YAML
apiVersion: config.nri/v1alpha1
kind: BalloonsPolicy
metadata:
  annotations:
    meta.helm.sh/release-name: controller-logs
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2024-11-07T12:46:22Z"
  generation: 2
  labels:
    app.kubernetes.io/instance: controller-logs
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: nri-resource-policy-balloons
    helm.sh/chart: nri-resource-policy-balloons-v0.0.0
  name: default
  namespace: kube-system
  resourceVersion: "5660"
  uid: 2f144427-c400-469e-999c-f8a0bf476101
spec:
  allocatorTopologyBalancing: true
  balloonTypes:
  - allocatorPriority: normal
    memoryTypes:
    - foo
    - bar
    minBalloons: 0
    minCPUs: 1
    name: default
    namespaces:
    - default
    preferIsolCpus: false
    shareIdleCPUsInSame: system

```

Reply:
@fmuyassarov Are you sure you really have that CRD in place. Just asking to make sure you didn't try to helm install and helm silently refused to replace an existing version without that validation pattern in place, and that's why it does not work...

Reply:
I can try to delete and install crds too to double check. I must say I am not 100% sure. Will update soon.

Reply:
You were right, I was still using the old CRD. 
But at least we should consider the next comment below to avoid this

```
    memoryTypes:
    - HBM
    - HBM

```

Reply:
Now handled with listType=set